cardano_node_tests/tests/test_pools.py

```diff
@@ -87,6 +88,108 @@ def cluster_mincost(
     )
 
 
+def _check_pool_deregistration_in_db(stake_pool_id: str, retiring_epoch: int):
+    """Check pool retirement in db-sync."""
+    pool_data_in_db = dbsync_utils.get_pool_data(stake_pool_id)
+
+    if pool_data_in_db.retire_announced_tx_id and pool_data_in_db.retiring_epoch:
+        assert (
+            retiring_epoch == pool_data_in_db.retiring_epoch
+        ), f"Mismatch in epoch values: {retiring_epoch} VS {pool_data_in_db.retiring_epoch}"
+    else:
+        raise AssertionError(f"Stake pool `{stake_pool_id}` not retired.")
+
+
+def _check_pool_data_in_db(
+    ledger_pool_data: dict, db_pool_data: dbsync_utils.PoolDataRecord
+) -> str:
+    """Check comparison for pool data between ledger and db-sync."""
+    # pylint: disable=too-many-statements,too-many-locals,too-many-branches
+    errors_list = []
+
+    if ledger_pool_data["publicKey"] != db_pool_data.hash:
+        errors_list.append(
+            "'publicKey' value is different than expected; "
+            f"Expected: {ledger_pool_data['publicKey']} vs Returned: {db_pool_data.hash}"
+        )
+
+    if ledger_pool_data["cost"] != db_pool_data.fixed_cost:
+        errors_list.append(
+            "'cost' value is different than expected; "
+            f"Expected: {ledger_pool_data['cost']} vs Returned: {db_pool_data.fixed_cost}"
+        )
+
+    metadata = ledger_pool_data.get("metadata") or {}
+
+    if metadata["hash"] and metadata["url"]:
+        metadata_hash = metadata["hash"]
+        if metadata_hash != db_pool_data.metadata_hash:
+            errors_list.append(
+                "'metadata hash' value is different than expected; "
+                f"Expected: {metadata_hash} vs "
+                f"Returned: {db_pool_data.metadata_hash}"
+            )
+
+        metadata_url = metadata["url"]
+        if metadata_url != db_pool_data.metadata_url:
+            errors_list.append(
+                "'metadata url' value is different than expected; "
+                f"Expected: {metadata_url} vs "
+                f"Returned: {db_pool_data.metadata_url}"
+            )
+
+    elif ledger_pool_data["metadata"] is not None:
+        errors_list.append(
+            "'metadata' value is different than expected; "
+            f"Expected: None vs Returned: {ledger_pool_data['metadata']}"
+        )
+
+    if {*ledger_pool_data["owners"]} != db_pool_data.owners:
```

Comment:
I "wrapped" it in `sorted` in order to avoid failures due to ordering pf list elements:

```python
if sorted(ledger_pool_data["owners"]) != sorted(db_pool_data.owners):
```